### PR TITLE
Tw/snippets/desc

### PR DESCRIPTION
**Pull request by imtayadeway**

### CODE REVIEW

**Comment by dgutov** [comment]
Thanks. I think I may have been to hasty with the suggestion, though.

Even if `expose_dsl_globally` is not set to false, starting with RSpec 3 (or 2.99?), we can prepend `RSpec.` to the top-level `describe` anyway, right?

Then, maybe the right approach is to have the user option anyway and probably ignore the contents of `spec_helper.rb`. As a bonus, you can look inside `Gemfile.lock` and check that the version of `rspec` is high enough.

File: rspec-mode.el [hunk 1]
```diff
@@ -755,6 +755,13 @@ or a cons (FILE . LINE), to run one example."
         (concat (rspec-project-root) "spec/spec_helper.rb"))
        (re-search-forward "include +FactoryGirl::Syntax::Methods" nil t)))))
 
+(defun rspec-expose-dsl-globally-p ()
+  "Check whether expose_dsl_globally is set to false in spec_helper."
+  (with-temp-buffer
+    (insert-file-contents
+     (concat (rspec-project-root) "spec/spec_helper.rb"))
```
Comment (dgutov) [inline rspec-mode.el]: The usual idiom is to use `expand-file-name` and pass the directory as the second argument.

**Comment by pezra** [comment]
I agree with @dgutov, it would be better to skip the `spec_helper.rb` file inspection and always include the `RSpec.` leader at the top level by default. We would need an customization variable to allow the user to disable the leader it if they are using an incompatible version of rspec.

Checking the `Gemfile.lock` would fine but unnecessary, imo.

**Comment by imtayadeway** [comment]
thanks for the comments! I'll do some work on this tonight

**Comment by imtayadeway** [comment]
Just updated my PR. I chose not to do any Gemfile.lock checking in this iteration to keep things simple.

**Comment by dgutov** [comment]
Looks good. Could you squash the last two commits and force-push?

**Comment by imtayadeway** [comment]
@dgutov done!

**Comment by dgutov** [comment]
Thanks!

**Pull request merged**